bigquery/system-test/datasets.test.js

```diff
@@ -1,94 +1,61 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 
 'use strict';
 
-var BigQuery = require('@google-cloud/bigquery');
-var uuid = require('node-uuid');
-var program = require('../datasets');
+const bigquery = require(`@google-cloud/bigquery`)();
+const uuid = require(`node-uuid`);
+const path = require(`path`);
+const run = require(`../../utils`).run;
 
-var bigquery = BigQuery();
-var projectId = process.env.GCLOUD_PROJECT;
-var datasetId = 'nodejs-docs-samples-test-' + uuid.v4();
+const cwd = path.join(__dirname, `..`);
+const cmd = `node datasets.js`;
+const datasetId = (`nodejs-docs-samples-test-${uuid.v4()}`).replace(/-/gi, '_');
 
-// BigQuery only accepts underscores
-datasetId = datasetId.replace(/-/gi, '_');
+describe(`bigquery:datasets`, function () {
+  after(() => bigquery.dataset(datasetId).delete({ force: true }).catch(() => undefined));
 
-describe('bigquery:datasets', function () {
-  after(function (done) {
-    bigquery.dataset(datasetId).delete({
-      force: true
-    }, function () {
-      // Ignore any error, the dataset might already have been successfully deleted
-      done();
-    });
-  });
-
-  describe('createDataset', function () {
-    it('should create a new dataset', function (done) {
-      program.createDataset(datasetId, function (err, dataset, apiResponse) {
-        assert.equal(err, null);
-        assert.notEqual(dataset, undefined);
-        assert.equal(dataset.id, datasetId);
-        assert.equal(console.log.calledOnce, true);
-        assert.deepEqual(console.log.firstCall.args, ['Created dataset: %s', datasetId]);
-        assert.notEqual(apiResponse, undefined);
-
-        done();
-      });
-    });
+  it(`should create a dataset`, () => {
+    const output = run(`${cmd} create ${datasetId}`, cwd);
+    assert.equal(output, `Dataset ${datasetId} created.`);
+    return bigquery.dataset(datasetId).exists()
+      .then((results) => assert.equal(results[0], true));
   });
 
-  describe('listDatasets', function () {
-    it('should list datasets', function (done) {
-      program.listDatasets(projectId, function (err, datasets) {
-        assert.equal(err, null);
-        assert.equal(Array.isArray(datasets), true);
-        assert.equal(datasets.length > 0, true);
-        var matchingDatasets = datasets.filter(function (dataset) {
-          return dataset.id === datasetId;
-        });
-        assert.equal(matchingDatasets.length, 1, 'newly created dataset is in list');
-        assert.equal(console.log.calledOnce, true);
-        assert.deepEqual(console.log.firstCall.args, ['Found %d dataset(s)!', datasets.length]);
-
-        done();
-      });
-    });
+  it(`should list datasets`, (done) => {
+    // Listing is eventually consistent. Give the indexes time to update.
+    setTimeout(() => {
+      const output = run(`${cmd} list`, cwd);
+      assert.notEqual(output.indexOf(`Datasets:`), -1);
```

Comment:
Use `includes()` instead of `indexOf()`? (If so, apply this comment through the rest of your PR.)